src/components/remote_control/src/resource_allocation_manager_impl.cc

```diff
@@ -0,0 +1,215 @@
+#include "remote_control/resource_allocation_manager_impl.h"
+#include "application_manager/application.h"
+#include "application_manager/message_helper.h"
+#include "remote_control/rc_module_constants.h"
+#include "json/json.h"
+#include "utils/helpers.h"
+#include "utils/make_shared.h"
+#include "remote_control/message_helper.h"
+
+namespace remote_control {
+
+CREATE_LOGGERPTR_GLOBAL(logger_, "RemoteControlModule")
+
+ResourceAllocationManagerImpl::ResourceAllocationManagerImpl(
+    RemotePluginInterface& rc_plugin)
+    : current_access_mode_(hmi_apis::Common_RCAccessMode::AUTO_ALLOW)
+    , rc_plugin_(rc_plugin) {}
+
+ResourceAllocationManagerImpl::~ResourceAllocationManagerImpl() {}
+
+AcquireResult::eType ResourceAllocationManagerImpl::AcquireResource(
+    const std::string& module_type, const uint32_t app_id) {
```

Comment:
Why use module_type as string instead of enums defined by hmi or mobile api?

Reply:
@JackLivio good issue. I will check if it possible to use API enum. Main idea that module types also stored in policy table as strings and in messages from HMI SDL receive it as strings in json.

Reply:
@JackLivio issue is that it is not clear - using mobile of HMI API. And there can me implicit problems in case if this enums won't match each other as integers (we had big experience in debugging problems, when root cause was casting mobile to hmi API, and it was not synchronized). So I think that it is better solution to use strings here.